src/client/opamState.ml

```diff
@@ -1989,7 +1989,7 @@ let string_of_env_update t shell updates =
       | "+="
       | ":=" -> (ident, Printf.sprintf "%s:$%s" string ident)
       | "=:"
-      | "=+" -> (ident, Printf.sprintf "$%s:%s" ident string)
+      | "=+" -> (ident, Printf.sprintf (match shell with `csh -> "${%s}:%s" | _ -> "$%s:%s") ident string)
```

Comment:
The reason for this change is that (t)csh raises an _error_ for "$MANPATH:..." since ":" is interpreted as a modifier for the variable reference "$MANPATH".

However, it fails to type-check on OCaml 3.12.1 (sorry again!).  A possible alternative would be:

(match shell with `csh -> Printf.sprintf "${%s}:%s" ident string | _ -> Printf.sprintf "$%s:%s" ident string)